packages/amplify_auth_cognito/ios/Classes/FlutterSignUpRequest.swift

```diff
@@ -13,41 +13,51 @@
  * express or implied. See the License for the specific language governing
  * permissions and limitations under the License.
  */
- 
+
 import Foundation
 import Amplify
 import AmplifyPlugins
 import amplify_core
 
 struct FlutterSignUpRequest {
-  var username: String
-  var password: String
-  var userAttributes: [AuthUserAttribute] = []
-  var options: Dictionary<String, Any>? = [:]
-  init(dict: NSMutableDictionary){
-    self.username = dict["username"] as! String
-    self.password = dict["password"] as! String
-    self.options = dict["options"] as? Dictionary<String, Any>
-    self.userAttributes = self.createAuthUserAttributes(options: dict["options"] as! Dictionary<String, Any>)
-  }
+    var username: String
+    var password: String
+    var options: AuthSignUpRequest.Options?
+    init(dict: NSMutableDictionary){
+        self.username = dict["username"] as! String
+        self.password = dict["password"] as! String
+        self.options = createOptions(options: dict["options"] as? Dictionary<String, Any>)
+    }
+    
+    private func createOptions(options: Dictionary<String, Any>?) -> AuthSignUpRequest.Options? {
+        if (options == nil) {
+            return nil
+        }
+        var formattedAttributes: Array<AuthUserAttribute> = Array()
+        if (options!["userAttributes"] is Dictionary<String, String>) {
+            let rawAttributes: Dictionary<String, Any> = options!["userAttributes"] as! Dictionary<String, String>
+            for attr in rawAttributes {
+                formattedAttributes.append(createAuthUserAttribute(key: attr.key, value: attr.value as! String))
+            }
+        }
+        return AuthSignUpRequest.Options(userAttributes: formattedAttributes)
+        
+    }
     
-  func createAuthUserAttributes(options: Dictionary<String, Any>) -> [AuthUserAttribute] {
-    let rawAttributes: Dictionary<String, Any> = options["userAttributes"] as! Dictionary<String, String>
-    var formattedAttributes: Array<AuthUserAttribute> = Array()
-    for attr in rawAttributes {
-        formattedAttributes.append(createAuthUserAttribute(key: attr.key, value: attr.value as! String))
+    private func createAuthUserAttributes(options: Dictionary<String, Any>) -> [AuthUserAttribute] {
+        let rawAttributes: Dictionary<String, Any> = options["userAttributes"] as! Dictionary<String, String>
+        var formattedAttributes: Array<AuthUserAttribute> = Array()
+        for attr in rawAttributes {
+            formattedAttributes.append(createAuthUserAttribute(key: attr.key, value: attr.value as! String))
+        }
+        return formattedAttributes
```

Comment:
Nit: Could simplify this I think
```suggestion
        let rawAttributes: Dictionary<String, Any> = options["userAttributes"] as! Dictionary<String, String>
        return rawAttributes.map { attr in
            createAuthUserAttribute(key: attr.key, value: attr.value as! String)
        }
```